server/src/main/java/org/elasticsearch/index/seqno/RetentionLeases.java

```diff
@@ -136,7 +136,25 @@ public RetentionLeases(final long primaryTerm, final long version, final Collect
         Objects.requireNonNull(leases);
         this.primaryTerm = primaryTerm;
         this.version = version;
-        this.leases = Collections.unmodifiableMap(toMap(leases));
+        final int leasesCount = leases.size();
+        switch (leasesCount) {
+            case 0 -> this.leases = Map.of();
+            case 1 -> {
+                final RetentionLease lease = leases.get(0);
+                this.leases = Map.of(lease.id(), lease);
+            }
+            default -> {
+                // use a linked hash map to preserve order
+                final LinkedHashMap<String, RetentionLease> map = Maps.newLinkedHashMapWithExpectedSize(leasesCount);
+                for (RetentionLease lease : leases) {
+                    map.merge(lease.id(), lease, (left, right) -> {
```

Comment:
Any reason we don't just use `Map#put` here and throw the `ISE` if it doesn't return `null`?

Reply:
Right ... I guess I just mindlessly refactored from the existing code and liked how short this was :) put and throw it is, likely faster this way.